packages/peregrine/lib/talons/CartPage/ProductListing/useProduct.js

```diff
@@ -109,10 +110,18 @@ export const useProduct = props => {
         return null;
     }, [formatMessage, showLoginToast]);
 
+    const handleWishlistUpdateError = useCallback(() => {
+        setIsCartUpdating(true);
+
+        fetchCartDetails({ variables: { cartId } });
```

Comment:
Threading this prop all the way down from the cart page feels odd. Could you tell me why this is necessary in the first place and how to repro or get this function to be called? If we do have to go this route, perhaps we can go with a callback like `onError` that is passed down.

Reply:
The react state update error I showed earlier today was present even before this change. That said, you can reproduce an error that can call this function by opening the cart in 2 tabs, move an item in 1 cart to the wishlist and do the same on the other tab. Removing items from the cart will fail because the item is not in the cart anymore.

Reply:
Ok thanks for providing repro.

I do think the handling here is a bit off still - the item greys out, then the success toast pops up, then you see an error on the item for a brief moment and then the item is deleted from the cart page UI.

[![Image from Gyazo](https://i.gyazo.com/1a4ded3a8088a9d19680006f5d0f0c4f.gif)](https://gyazo.com/1a4ded3a8088a9d19680006f5d0f0c4f)

Reply:
Yeah, it is doggy for sure, but that's the best at the moment till GQL comes up with a unified query. And this is a very very edge case. UX approves this as a valid solution given the circumstances.

Reply:
> If this is the desired approach (refetching the cart data on error) then I think a callback such as `onError` defined by the cart page and passed to any children who care, is preferable to a `fetchCartDetails` prop.

Considering `useWishlist` as a helper talon, I preferred to name it `onErrorX` instead because later when we move to a unified `useWishlist` helper solution it will be a straight swap.

Reply:
> I also think we can remove the `setDisplayError(true)` call for the product causing the error since it's going to be deleted anyways.

Wouldn't that hide legitimate errors as well?

Reply:
> Wouldn't that hide legitimate errors as well?

Would it? Normally you would call `setDisplayError(true)` in the [catch block](https://github.com/magento/pwa-studio/blob/revanth/add-cart-to-wishlist/packages/peregrine/lib/talons/CartPage/ProductListing/useProduct.js#L197) related to the mutation that fails. Is this what we use this function for?

Perhaps it would hide legitimate errors but I still think the UX is weird. Look at the gif!

Reply:
I think I understand my confusion about this. In these files we have called `setDisplayError` in the `catch` block of the related item call, such as [here](https://github.com/magento/pwa-studio/pull/3130/files#diff-935be6164790da836b0c844633f2a602ae4ced096ff8bfc17c4427d2a13d885bR197). In fact, I believe that is the code that is causing the error to display, not what you added here on line 118.

So please disregard my suggestion to remove the call to `setDisplayError`. We do need it, in the event that the wishlist mutation throws an error.

I still would like to see my other concern addressed :)

Reply:
@revanth0212 had reasoning for not making my suggested change, I wanted to post it here so it was captured:

> I would prefer it called fetchCartDetails because form the cart page POV we are not using that function for all errors. We are only using that when the wishlist move errors out.

This makes sense to me.